cliquet/views/errors.py

```diff
@@ -67,9 +67,14 @@ def page_not_found(request):
                                    len(request.path)):]
         redirect = '%s%s' % (path, querystring)
         return HTTPTemporaryRedirect(redirect)
-    elif request.path.startswith('/' + request.registry.route_prefix):
+    elif request.path.startswith('/' + request.registry.route_prefix + '/'):
         errno = ERRORS.MISSING_RESOURCE
         error_msg = "The resource you are looking for could not be found."
+    elif trailing_slash_redirection_enabled:
+        querystring = request.url[(request.url.rindex(request.path) +
+                                   len(request.path)):]
+        redirect = '/%s/%s' % (request.registry.route_prefix, querystring)
+        return HTTPTemporaryRedirect(redirect)
 
     response = http_error(httpexceptions.HTTPNotFound(),
```

Comment:
There is apparently one case where you do not enter any ifs:

    request.path.startswith('/' + request.registry.route_prefix) and \
    not request.path.startswith('/' + request.registry.route_prefix + '/') and \
    not trailing_slash_redirection_enabled

In that specific case we do not define ``errno`` and ``error_msg``

Either I can change the ``elif`` on line 73 with a ``else`` making the redirection for ``/v1`` to ``/v1/`` even when trailing slash redirection is not enabled or I can add a else with ``errno`` et ``error_msg``.